### PR TITLE
Implement NFS Zone Settings Resource

### PR DESCRIPTION
**Pull request by MangirishK**
# Description
NFS Zone Settings Resource Implementation

# ISSUE TYPE
- Feature Pull Request

##### RESOURCE NAME
NFS Zone Settings

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by baoy1** [comment]
@MangirishK , please squash the 7 commits to one, and make sure your branch is based on the latest main branch before creating PR.

File: powerscale/provider/nfs_zone_settings_resource.go [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+
+	tflog.Info(ctx, "Completed deleting nfs zone settings")
+}
+
+// ImportState imports the resource.
+func (r *NfsZoneSettingsResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	tflog.Info(ctx, "Started importing nfs zone settings")
+
+	reqID := req.ID
+	zone := strings.Trim(reqID, " ")
```
Comment (baoy1) [inline powerscale/provider/nfs_zone_settings_resource.go:340]: strings.TrimSpace

File: powerscale/provider/nfs_zone_settings_resource.go [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	zone := plan.Zone.ValueString()
+
+	var toUpdate powerscale.V2NfsSettingsZoneSettings
+	err := helper.ReadFromState(ctx, &plan, &toUpdate)
+	if err != nil {
+		errStr := constants.UpdateNfsZoneSettingsErrorMsg + "with error: "
```
[the remark this answers is not in the document]
Comment (MangirishK) [reply powerscale/provider/nfs_zone_settings_resource.go]: addressed with "ReadNfsZoneSettingsErrorMsg"

File: powerscale/provider/nfs_zone_settings_resource.go (hunk 1 above)
Comment (MangirishK) [reply powerscale/provider/nfs_zone_settings_resource.go]: addressed with "TrimSpace"

**Comment by MangirishK** [comment]
Please find the screenshots here for acceptance testing and coverage:

Screenshot_1
Screenshot_2
Screenshot_3

**Comment by MangirishK** [comment]
run e2e test

**Comment by P-Cao** [comment]
run e2e test

File: powerscale/models/nfs_zone_settings.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
+/*
+Copyright (c) 2023-2024 Dell Inc., or its subsidiaries. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (MangirishK) [reply powerscale/models/nfs_zone_settings.go]: Updated the copyright year to 2024

File: powerscale/provider/nfs_zone_settings_resource.go [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+	state.Zone = plan.Zone
+	state.ID = plan.Zone
+
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Info(ctx, "Completed reading nfs zone settings")
```
[the remark this answers is not in the document]
Comment (MangirishK) [reply powerscale/provider/nfs_zone_settings_resource.go]: Updated the logs at the start and end both to have creating instead of reading

**Comment by taohe1012** [comment]
> @MangirishK , please squash the 7 commits to one, and make sure your branch is based on the latest main branch before creating PR.

@MangirishK please squash commits to one, thanks

**Comment by MangirishK** [comment]
run e2e test

**Review by taohe1012 — APPROVED** [review]

**Comment by forrestxia** [comment]
run e2e test

**Comment by MangirishK** [comment]
run e2e test

**Review by NavisJ — APPROVED** [review]

**Review by P-Cao — APPROVED** [review]

**Pull request merged**